mobile/examples/phi-3/android/app/src/main/java/ai/onnxruntime/genai/demo/MainActivity.java

```diff
@@ -160,17 +202,34 @@ private void downloadModels(Context context) throws GenAIException {
                 // so assuming if one filename exists, then the download model step has already
                 // be
                 // done.
-                genAIWrapper = createGenAIWrapper();
-                break;
+                if (index == urlFilePairs.size() - 1) {
+                    model = new Model(getFilesDir().getPath());
+                    tokenizer = model.createTokenizer();
+                    break;
+                }
+                continue;
             }
             executor.execute(() -> {
                 ModelDownloader.downloadModel(context, url, fileName, new ModelDownloader.DownloadCallback() {
+                    private long pctDone = 0;
+                    @Override
+                    public void onDownloadProgress(long bytesDone, long bytesTotal) {
+                        if (bytesTotal > 0) {
+                            long newPctDone = bytesDone * 100 / bytesTotal;
+                            if (newPctDone > pctDone) {
+                                pctDone = newPctDone;
+                                Log.d(TAG, "Download" + fileName + ": " + pctDone
+                                    + "% of " + (bytesTotal/1024) + " KB");
+                            }
```

Comment:
this seems like it could be simpler. I think ModelDownloader won't call this twice with the same bytesDone value so you may not need the newPctDone > pctDone check. if all we do is log, then maybe it is simpler to just do that in ModelDownloader.